azure-mgmt-monitor/src/main/java/com/microsoft/azure/management/monitor/implementation/ActivityLogsImpl.java

```diff
@@ -0,0 +1,274 @@
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithExecute filterByResource(String resourceId) {
+        this.filterString = String.format(" and resourceUri eq '%s'", resourceId);
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithExecute filterByResourceProvider(String resourceProviderName) {
+        this.filterString = String.format(" and resourceProvider eq '%s'", resourceProviderName);
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithExecute filterByCorrelationId(String correlationId) {
+        this.filterString = String.format(" and correlationId eq '%s'", correlationId);
+        return this;
+    }
+
+    @Override
+    public PagedList<EventData> execute() {
+        return listEventData(getOdataFilterString() + this.filterString);
+    }
+
+    @Override
+    public Observable<EventData> executeAsync() {
+        return listEventDataAsync(getOdataFilterString() + this.filterString);
+    }
+    private String getOdataFilterString() {
+        return String.format("eventTimestamp ge '%s' and eventTimestamp le '%s'",
+                this.queryStartTime.withZone(DateTimeZone.UTC).toString(ISODateTimeFormat.dateTime()),
+                this.queryEndTime.withZone(DateTimeZone.UTC).toString(ISODateTimeFormat.dateTime()));
+    }
+
+    private PagedList<EventData> listEventData(String filter) {
+        ArrayList<String> sortedList = new ArrayList<String>(this.responsePropertySelector);
+        Collections.sort(sortedList);
+        return (new PagedListConverter<EventDataInner, EventData>() {
+            @Override
+            public Observable<EventData> typeConvertAsync(EventDataInner inner) {
+                return Observable.just((EventData) new EventData(inner));
+            }
+        }).convert(this.inner().activityLogs().list(filter, StringUtils.join(sortedList, ',')));
+    }
+
+    private Observable<EventData> listEventDataAsync(String filter) {
+        ArrayList<String> sortedList = new ArrayList<String>(this.responsePropertySelector);
+        Collections.sort(sortedList);
+        return this.inner().activityLogs().listAsync(filter, StringUtils.join(sortedList, ','))
+                .flatMap(new Func1<Page<EventDataInner>, Observable<EventData>>() {
+                    @Override
+                    public Observable<EventData> call(Page<EventDataInner> eventDataInnerPage) {
+                        return Observable.from(eventDataInnerPage.items())
+                                .map(new Func1<EventDataInner, EventData>() {
+                                    @Override
+                                    public EventData call(EventDataInner eventDataInner) {
+                                        return new EventData(eventDataInner);
+                                    }
+                                });
+                    }
+                });
```

Comment:
Another way of doing this is by using `flatMapIterable` operator, same number of lines but reduce func nesting.

```java
return this.inner().activityLogs().listAsync(filter, StringUtils.join(sortedList, ','))
    .flatMapIterable(new Func1<Page< EventDataInner>, Iterable< EventDataInner>>() {
            @Override
            public Iterable<EventDataInner> call(Page<EventDataInner> page) {
                return page.items();
            }
        }).map(new Func1<EventDataInner, EventData>() {
            @Override
            public EventData call(EventDataInner eventDataInner) {
                return new EventData(eventDataInner);
            }
        });

```